### PR TITLE
Herwig7 81x

### PR DESCRIPTION
**Pull request by smuzaffar**

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: herwigpp-toolfile.spec [hunk 1]
```diff
@@ -11,12 +11,16 @@ cat << \EOF_TOOLFILE >%i/etc/scram.d/herwigpp.xml
 <tool name="herwigpp" version="@TOOL_VERSION@">
   <client>
     <environment name="HERWIGPP_BASE" default="@TOOL_ROOT@"/>
-    <environment name="LIBDIR" default="$HERWIGPP_BASE/lib"/>
-    <environment name="INCLUDE" default="$HERWIGPP_BASE/include"/>
+    <environment name="LIBDIR" default="$HERWIGPP_BASE/lib/Herwig7"/>
+    <environment name="INCLUDE" default="$HERWIGPP_BASE/include/Herwig7"/>
```
Comment (mharrend) [inline herwigpp-toolfile.spec]: @smuzaffar I am sorry it seems that I uploaded not the newest version of the tool file.

All parts with Herwig7 have to be changed to just Herwig as we found out in the old PR.
<environment name="LIBDIR" default="$HERWIGPP_BASE/lib/Herwig"/>
<environment name="INCLUDE" default="$HERWIGPP_BASE/include/Herwig"/>
<runtime name="HERWIGPATH" value="$HERWIGPP_BASE/share/Herwig"/>
Sorry for the additional efforts required. If you want and have time this week I will buy you a beer at the CERN R1.

Comment (smuzaffar) [reply herwigpp-toolfile.spec]: this is your original commit where you had Herwig7. The fix is part of another commit. If you see the final changes (by going to File Changed tab then you will see the right path)

**Comment by mharrend** [comment]
It seems that the madgraphamcatnlo-toolfile.spec file is missing and was not copied to this PR.
Furthermore, the issue with the ThePEG interface is expected since the interface has to be rewritten due to the new Herwig7 version. Can we deactivate the interface for now?

**Comment by smuzaffar** [comment]
you mean   madgraph5amcatnlo-toolfile.spec ? I do not see any changes for this file in your  repo https://github.com/mharrend/cmsdist/blob/IB/CMSSW_8_1_X/gcc530/madgraph5amcatnlo-toolfile.spec

**Comment by smuzaffar** [comment]
no idea about deactivating ThePEG interface ..... if there are some cmssw changes needed to go with this PR then please make a PR for cmssw repo too

**Comment by mharrend** [comment]
Okay, I have checked the exact error why the compilation fails:
`>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/cmsdist-test-pr/CMSSW_8_1_X_2016-08-30-2300/src/GeneratorInterface/ThePEGInterface/src/ThePEGInterface.cc 

> > Compiling  /build/cmsbuild/jenkins-workarea/workspace/cmsdist-test-pr/CMSSW_8_1_X_2016-08-30-2300/src/GeneratorInterface/ThePEGInterface/src/RandomEngineGlue.cc 
> > /build/cmsbuild/jenkins-workarea/workspace/cmsdist-test-pr/CMSSW_8_1_X_2016-08-30-2300/src/GeneratorInterface/ThePEGInterface/src/RandomEngineGlue.cc:5:49: fatal error: ThePEG/Interface/ClassDocumentation.h: No such file or directory
> > compilation terminated.
> > /build/cmsbuild/jenkins-workarea/workspace/cmsdist-test-pr/CMSSW_8_1_X_2016-08-30-2300/src/GeneratorInterface/ThePEGInterface/src/ThePEGInterface.cc:21:44: fatal error: ThePEG/Utilities/DynamicLoader.h: No such file or directory
> > compilation terminated.`

The issue is that the two files ThePEG/Utilities/DynamicLoader.h and ThePEG/Interface/ClassDocumentation.h are not found which are referenced by
`#include <ThePEG/Utilities/DynamicLoader.h>`
and 
`#include <ThePEG/Interface/ClassDocumentation.h>`

Thus, it seems that the include path <ThePEG/...> is not properly set / propagated.

There are now 2 solutions:
1.  We can temporarily deactivate the ThePEG interface which has to be rewritten /refactored anyway soon.
2.  We could check now why the ThePEG include path is not working, but it can happen that we get then later some compilation error if the ThePEG library functions were changed.

What would you propose? I will provide a PR to remove ThePEG from CMSSW, so that we can merge Herwig7 and rewrite the ThePEG interface.

**Comment by mharrend** [comment]
If have now created the following PR in CMSSW https://github.com/cms-sw/cmssw/pull/15704 which removes the deprecated ThePEG interface, afterwards the tests can be started again and will succeed.
@smuzaffar Can you please merge the CMSSW PR and then start the tests.

Please note that the deprecated ThePEG interface will be replaced by a Herwig7 interface which has to be partly rewritten anyway.

**Comment by ghost** [comment]
please test
(let's try again...)

**Comment by smuzaffar** [comment]
@degano , this should be tested with cms-sw/cmssw#15704
so tests need to be triggered from cmssw PR

**Comment by smuzaffar** [comment]
assign generators

**Comment by smuzaffar** [comment]
+externals

**Comment by govoni** [comment]
+1

**Comment by smuzaffar** [comment]
cms-sw/cmssw#15704 is still open. This external should go along with the cmssw change.
@davidlange6 , if no objection then you can merge cms-sw/cmssw#15704

**Comment by mharrend** [comment]
Dear @smuzaffar, dear all,

unfortunately due to some path issue Herwig7 is still not working in CMSSW. I have created an issue report to track this bug #2571.

**Pull request merged**